pkg/providers/ovirt/mappings/merger.go

```diff
@@ -0,0 +1,83 @@
+package mappings
+
+import (
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	"github.com/kubevirt/vm-import-operator/pkg/utils"
+)
+
+// MergeMappings creates new resource mapping spec containing all of the mappings from the externalMappingSpec with all mappings from crMappings. Mappings from the crMappings will overwrite ones from the external mapping.
+func MergeMappings(externalMappingSpec *v2vv1alpha1.ResourceMappingSpec, vmiMapping *v2vv1alpha1.OvirtMappings) v2vv1alpha1.OvirtMappings {
+	if externalMappingSpec == nil && vmiMapping == nil {
+		return v2vv1alpha1.OvirtMappings{}
+	}
+	primaryMappings, secondaryMappings := extractMappings(externalMappingSpec, vmiMapping)
+
+	networkMappings := mergeMappings(primaryMappings.NetworkMappings, secondaryMappings.NetworkMappings)
+	storageMappings := mergeMappings(primaryMappings.StorageMappings, secondaryMappings.StorageMappings)
```

Comment:
My only comment regarding this - the merged mappings might contain more entries than what is actually needed for the VM import. However, i think the fee is small, comparing how complex this could look if we'll throw in the source VM.

We need to make sure to consider only the resources which are relevant for the VM Import on the validation (e.g. not fail if a network that isn't required by the VMI doesn't exist)

Reply:
That's how the mapping validator behaves - I will add test case to show that it works.